lib/Config.js

```diff
@@ -280,6 +280,39 @@ function requestsConfigAssert(requestsConfig) {
     }
 }
 
+function bucketNotifAssert(bucketNotifConfig) {
+    const { resource, type, host, port, auth } = bucketNotifConfig;
+    assert(typeof resource === 'string',
+        'bad config: bucket notification configuration resource must be a string');
+    assert(typeof type === 'string',
+        'bad config: bucket notification configuration type must be a string');
+    assert(typeof host === 'string' && host !== '',
+        'bad config: hostname must be a non-empty string');
+    assert(Number.isInteger(port) && port > 0,
+        'bad config: port must be a positive integer');
+    assert(typeof auth === 'object',
+        'bad config: bucket notification auth must be an object');
+    const { cert, user, password } = auth;
```

Comment:
I am still figuring out auth part of the destinations. This may change in the future. For the time being this works.